atlasdb-impl-shared/src/main/java/com/palantir/atlasdb/sweep/metrics/SweepOutcomeMetrics.java

```diff
@@ -0,0 +1,85 @@
+/*
+ * Copyright 2018 Palantir Technologies, Inc. All rights reserved.
+ *
+ * Licensed under the BSD-3 License (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://opensource.org/licenses/BSD-3-Clause
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.palantir.atlasdb.sweep.metrics;
+
+import java.util.Arrays;
+import java.util.List;
+import java.util.concurrent.TimeUnit;
+
+import com.codahale.metrics.SlidingTimeWindowReservoir;
+import com.google.common.collect.ImmutableList;
+import com.google.common.collect.ImmutableMap;
+import com.palantir.atlasdb.AtlasDbMetricNames;
+import com.palantir.atlasdb.sweep.BackgroundSweeperImpl;
+import com.palantir.atlasdb.util.MetricsManager;
+
+public final class SweepOutcomeMetrics {
+    public static final List<SweepOutcome> LEGACY_OUTCOMES = Arrays.asList(SweepOutcome.values());
+    public static final List<SweepOutcome> TARGETED_OUTCOMES = ImmutableList.of(SweepOutcome.NOT_ENOUGH_DB_NODES_ONLINE,
+            SweepOutcome.DISABLED, SweepOutcome.SUCCESS, SweepOutcome.ERROR, SweepOutcome.NOTHING_TO_SWEEP);
+
+    private final SlidingTimeWindowReservoir reservoir;
+    private volatile boolean shutdown = false;
+    private volatile boolean fatal = false;
+
+    private SweepOutcomeMetrics() {
+        reservoir = new SlidingTimeWindowReservoir(60L, TimeUnit.SECONDS);
+    }
+
+    public static SweepOutcomeMetrics registerLegacy(MetricsManager metricsManager) {
+        SweepOutcomeMetrics metrics = new SweepOutcomeMetrics();
+        metrics.registerMetric(metricsManager, LEGACY_OUTCOMES, BackgroundSweeperImpl.class);
+        return metrics;
+    }
+
+    public static SweepOutcomeMetrics registerTargeted(MetricsManager metricsManager) {
+        SweepOutcomeMetrics metrics = new SweepOutcomeMetrics();
+        metrics.registerMetric(metricsManager, TARGETED_OUTCOMES, TargetedSweepMetrics.class);
+        return metrics;
+    }
+
+    private <T> void registerMetric(MetricsManager manager, List<SweepOutcome> outcomes, Class<T> forClass) {
```

Comment:
Do we have to register metrics on a class name?  We're finding this is quite brittle and expensive.  If we refactor some code we can inadvertently break our monitors and dashboards.

Reply:
Offline: decided not to change this at the moment, but should consider in the future.